changelog.d/5098.misc

```diff
@@ -0,0 +1 @@
+Add workarounds for pep-517 install errors.
```

Comment:
Was this intended to be included in this PR?

Reply:
Yes, this was pulled by cherry-picking https://github.com/matrix-org/synapse/pull/5115/commits/0cc91efd6af3c8009e2c243e7a77e8fc18f1e808 in order to prevent the CI from exploding.